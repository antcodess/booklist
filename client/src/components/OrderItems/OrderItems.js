import React from 'react';

import OrderItem from '../OrderItem/OrderItem';
import './OrderItems.scss';

export const OrderItems = ({ order }) => {
  const { id, total, products, createdAt } = order;
  const dateObject = new Date(createdAt.toDate());
  const month = dateObject.getMonth() + 1;
  const day = dateObject.getDate();
  const year = dateObject.getFullYear();
  const date = `${month}/${day}/${year}`;

  return (
    <div className='order-items'>
      <div className='order-details'>
        <span className='order-number'>
          <span className='bold'>Order #: </span>
          {id}
        </span>
        <span className='order-date'>
          <span className='bold'>Order Placed: </span>
          {date}
        </span>
      </div>
      <div className='order-header'>
        <div className='header-block'>
          <span>Product</span>
        </div>
        <div className='header-block'>
          <span>Title</span>
        </div>
        <div className='header-block'>
          <span>Quantity</span>
        </div>
        <div className='header-block'>
          <span>Price</span>
        </div>
      </div>
      { products.map(item => <OrderItem key={item.id} item={item} />)}
      <div className='total'>
        <span className='total-bold'>Total: <span>${total}</span></span>
      </div>
    </div>
  );
};

export default OrderItems;