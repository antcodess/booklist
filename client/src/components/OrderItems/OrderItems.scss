.order-items {
  width: 100%;

  .order-header {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr 1fr;
    grid-gap: 10px;
    padding-bottom: 20px;
    border-bottom: 2px solid #ededed;
    text-align: center;
    
    .header-block {
      span {
        font-family: "Hurme Geometric Sans 3 W03 Bold";
      }
    }
    
  }
  .total {
    border-top: 2px solid #ededed;
    display: flex;
    justify-content: flex-end;
    padding-top: 10px;
    margin-top: 40px;
    .total-bold {
      font-family: "Hurme Geometric Sans 3 W03 Bold";
    }
  }
}