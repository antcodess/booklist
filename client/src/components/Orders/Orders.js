import React from 'react';
import { Link } from 'react-router-dom';

import OrderItems from '../../components/OrderItems/OrderItems';

const Orders = ({ orderItems }) => {
  return (
    <React.Fragment>
      { orderItems.length ?
        orderItems.map(order => <OrderItems key={order.id} order={order} />)
        : (
            <div className='orders-empty-message'>
              <span>There are no orders. Let's change that. </span>
              <Link className='shop-link' to='/shop'>Shop Now</Link>
            </div>
          )
      }
    </React.Fragment>
  );
};

export default Orders;