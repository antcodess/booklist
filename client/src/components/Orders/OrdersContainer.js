import { connect } from 'react-redux';
import { compose } from 'redux';

import { createStructuredSelector } from 'reselect';

import { selectIsOrdersFetching } from '../../redux/orders/orders.selectors';
import Spinner from '../Spinner/Spinner';
import Orders from './Orders';

const mapStateToProps = createStructuredSelector({
  isLoading: selectIsOrdersFetching
});

const OrdersContainer = compose(
  connect(mapStateToProps),
  Spinner
)(Orders);

export default OrdersContainer;
