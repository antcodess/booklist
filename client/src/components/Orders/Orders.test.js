import React from 'react';
import { shallow } from 'enzyme';
import { Orders } from './Orders';
import OrderItems from '../OrderItems/OrderItems';

describe('Orders Component', () => {
  let wrapper;
  let wrapper2;
  let mockOrderItems;
  let dummyFunction = jest.fn();
  
  beforeEach(() => {
    
    mockOrderItems = [{id: 1}, {id: 2}, {id: 3}];

    const mockProps = {
      orderItems: mockOrderItems
    };

    const mockProps2 = {
      orderItems: []
    };
    
    // With Order Items
    wrapper = shallow(<Orders { ...mockProps } />);
    // Without Order Items
    wrapper2 = shallow(<Orders { ...mockProps2 } />);
  });

  it('should render Orders Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render same amount of OrderItems components as orderItems', () => {
    expect(wrapper.find(OrderItems).length).toEqual(mockOrderItems.length);
  });

  it('should not render OrderItems components if orderItems is empty', () => {
    expect(wrapper2.exists(OrderItems)).toBe(false);
  });

  it('should show empty message if orderItems are empty', () => {
    expect(wrapper2.exists('.orders-empty-message')).toBe(true);
  });

  it('should not show empty message if orderItems are empty', () => {
    expect(wrapper.exists('.orders-empty-message')).toBe(false);
  });
});
