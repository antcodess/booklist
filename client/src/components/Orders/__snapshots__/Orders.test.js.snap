// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders Component should render Orders Component 1`] = `ShallowWrapper {}`;
